src/JSInterop/Microsoft.JSInterop.JS/src/src/Microsoft.JSInterop.ts

```diff
@@ -61,6 +61,17 @@ export module DotNet {
     0: new JSObject(window, 0),
   };
 
+  cachedJSObjectsById[0]._cachedFunctions.set('import', (url: any) => {
+    // In most cases developers will want to resolve dynamic imports relative to the base HREF.
+    // However since we're the one calling the import keyword, they would be resolved relative to
+    // this framework bundle URL. Fix this by providing an absolute URL.
+    if (typeof url === 'string' && url.startsWith('./')) {
+      url = document.baseURI + url.substr(2);
+    }
+
+    return import(/* webpackIgnore: true */ url);
```

Comment:
The `webpackIgnore` thing is needed because otherwise, when we build `blazor.*.js`, Webpack will replace the `import` call with some special gunk of its own that tries to find Webpack modules instead of doing an actual native dynamic import.